zkvm/src/vm.rs

```diff
@@ -735,36 +734,24 @@ where
         Ok(())
     }
 
-    fn left_or_right<F>(&mut self, assign: F) -> Result<(), VMError>
-    where
-        F: FnOnce(&mut Contract, Predicate, Predicate) -> (),
-    {
-        let r = self.pop_item()?.to_data()?.to_predicate()?;
-        let l = self.pop_item()?.to_data()?.to_predicate()?;
+    fn select(&mut self, n: u8, k: u8) -> Result<(), VMError> {
+        if k >= n {
+            return Err(VMError::PredicateIndexInvalid)
+        }
 
+        let mut x_vec: Vec<Predicate> = Vec::with_capacity(n as usize);
```

Comment:
`x_vec` is not a helpful name. It's clear that it's a Vec, so maybe call it simply `preds`?